tests/unit/moduleapi/misc.tcl

```diff
@@ -248,28 +248,24 @@ start_server {tags {"modules"}} {
     test {rm_call EVAL - OOM} {
         r config set maxmemory 1
 
+        # OOM checking at RM_Call time, but not an allow-oom/no-writes script, so fails
         assert_error {OOM command not allowed when used memory > 'maxmemory'. script*} {
-            r test.rm_call eval {
+            r test.rm_call_flags M eval {
                 redis.call('set','x',1)
                 return 1
             } 1 x
         }
 
-        r test.rm_call eval {#!lua flags=no-writes
+        # OOM checking at RM_Call time, but a non deny oom script, so script runs
+        r test.rm_call_flags M eval {#!lua flags=no-writes
             redis.call('get','x')
             return 2
         } 1 x
 
-        assert_error {OOM allow-oom flag is not set on the script,*} {
-            r test.rm_call eval {#!lua
-                redis.call('get','x')
-                return 3
-            } 1 x
-        }
-
+        # no OOM checking at rm_call, so script runs
         r test.rm_call eval {
-            redis.call('get','x')
-            return 4
+            redis.call('set','x',1)
+            return 1
```

Comment:
i'd rather keep this test using SET` and call it with the M flag.
i.e. over all, this test block had 4 tests, validating the extraction of the shebang flags and their handling, i think i'd like to keep them as is, and just switch them all to use the M flag.

then Add two additional 4 tests **without** the M flag:
1. no shebang that does a GET and succeeds
2. no shebang that does a SET and is blocked
3. shebang with default flags that gets blocked
4. shebang with no-writes flag that does a get and succeeds.

Reply:
> then Add few additional 4 tests **without** the M flag:
> 
>     1. no shebang that does a GET and succeeds
> 
>     2. no shebang that does a SET and is blocked
> 
>     3. shebang with default flags that gets blocked
> 
>     4. shebang with no-writes flag that does a get and succeeds.

without M, they should all pass....

Reply:
sorry, don't know what went through my head.
so either we add all 4 tests and expect them all to pass, or maybe just add one or the two that are in theory blocked [2 and 3 above], and show that because the missing "M" flag they pass instead of failing.

Reply:
I think I'm unclear on the semantics we want, especially in the case of no M flag.

test cases

1) M - no shebang (get) - fail in rm_call
2) M - shebang default (get) - should fail in rm_call
3) M - shebang for allow-oom (set) - should pass
4) M - shebang for no-writes (get)- should pass
5) M - shebang for no-writes (get!) - should fail in script handling (i.e. get through rm_call)

can repeat all the above cases without M, they should all pass except #5 that should also fail in script handling

Reply:
for [5] you meant `no-writes` that uses **set**, right? not **get**.
we already have that test earlier, no need to repeat it in the OOM group and mess with the M flag;
```
    test {rm_call EVAL} {
        r test.rm_call eval {
            redis.call('set','x',1)
            return 1
        } 1 x

        assert_error {ERR Write commands are not allowed from read-only scripts.*} {
            r test.rm_call eval {#!lua flags=no-writes
                redis.call('set','x',1)
                return 1
            } 1 x
        }
    }
```

i pushed a cleanup commit, see commit comment for details